### PR TITLE
Substitute filename in bundle name

### PR DESCRIPTION
**Pull request by alexkuz**

### CODE REVIEW

**Comment by alexkuz** [comment]
@gaearon review plz :)

**Comment by gaearon** [comment]
Hmm... I don't get it :stuck_out_tongue:

**Comment by alexkuz** [comment]
That's how I use it:
```
loaders: {
      { test: /\.i18n\.json$/, loader: 'promise?global,[name].i18n' },
```

that way I can create separate bundle for each langpack file.

File: index.js [hunk 1]
```diff
@@ -10,6 +10,10 @@ module.exports.pitch = function (remainingRequest) {
   var query = this.query.substring(1).split(','),
     promiseLib = query[0],
     bundleName = query[1] || '';
+  var fileName = remainingRequest.match(/\/([^\/]+)$/)[1];
+  var name = fileName.split('.')[0];
```
Comment (gaearon) [inline index.js]: Can we use something like `path.basename()` instead of regexes?

Comment (alexkuz) [inline index.js]: @gaearon

File: index.js [hunk 2]
```diff
@@ -4,19 +4,24 @@
...
 
+ var path = require('path');
+
 module.exports = function () {};
 module.exports.pitch = function (remainingRequest) {
   this.cacheable && this.cacheable();
   var query = this.query.substring(1).split(','),
     promiseLib = query[0],
     bundleName = query[1] || '';
+  var file = path.parse(remainingRequest);
```
Comment (gaearon) [inline index.js]: AFAIK `path.parse` isn't available in 0.10 which some people still use. Better to use `path.basename` instead.

**Comment by alexkuz** [comment]
Here you go

**Comment by gaearon** [comment]
Out in 1.0, thanks!

**Pull request merged**